File: CS/DXSampleDistributedApplication/DXSampleWebService.asmx.cs
using System.Web.Services;
using System.ComponentModel;
using DevExpress.Xpo;
using DXSample.Service.Model;
using DevExpress.Data.Filtering;
using DevExpress.Xpo.Metadata;
using System;
using System.Collections.Generic;

namespace DXSample.Service {
    [WebService(Namespace = "http://www.devexpress.com/")]
    [WebServiceBinding(ConformsTo = WsiProfiles.BasicProfile1_1)]
    [ToolboxItem(false)]
    public class DXSampleWebService :WebService {
        [WebMethod]
        public string GetDataView (string className, ViewProperty[] properties, string filter) {
            using (Session session = new Session()) {
                XPClassInfo classInfo = session.GetClassInfo(typeof(Category).Assembly.FullName, 
                    string.Concat(typeof(Category).Namespace, ".", className));
                XPView result = new XPView(session, classInfo, new CriteriaOperatorCollection(), 
                    CriteriaOperator.Parse(filter));
                result.Properties.AddRange(properties);
                return new PersistentObjectToXmlConverter().ConvertViewToXml(result);
            }
        }
    }
}

File: VB/DXSampleDistributedApplication/DXSampleWebService.asmx.vb
Imports Microsoft.VisualBasic
Imports System.Web.Services
Imports System.ComponentModel
Imports DevExpress.Xpo
Imports DXSample.Service.Model
Imports DevExpress.Data.Filtering
Imports DevExpress.Xpo.Metadata
Imports System
Imports System.Collections.Generic

Namespace DXSample.Service
	<WebService(Namespace := "http://www.devexpress.com/"), WebServiceBinding(ConformsTo := WsiProfiles.BasicProfile1_1), ToolboxItem(False)> _
	Public Class DXSampleWebService
		Inherits WebService
		<WebMethod> _
		Public Function GetDataView(ByVal className As String, ByVal properties() As ViewProperty, ByVal filter As String) As String
			Using session As New Session()
				Dim classInfo As XPClassInfo = session.GetClassInfo(GetType(Category).Assembly.FullName, String.Concat(GetType(Category).Namespace, ".", className))
				Dim result As New XPView(session, classInfo, New CriteriaOperatorCollection(), CriteriaOperator.Parse(filter))
				result.Properties.AddRange(properties)
				Return New PersistentObjectToXmlConverter().ConvertViewToXml(result)
			End Using
		End Function
	End Class
End Namespace